### PR TITLE
Create Buggy Code fixed

### PR DESCRIPTION
**Pull request by rjagosto**

### CODE REVIEW

**Comment by hyowonshin** [comment]
# Buggy Code 
# Load data 
load(url("http://www.jakebowers.org/Data/meddat.rda"))
# Get names of columns
colnames(meddat)
# Put information into table 
table(Group, useNA = "ifany")
table(meddat$Group, useNA = "ifany"
# Find the mean of the population in 2003
mean(Pop2003)
mean(meddat$Pop2003)
# Remove 'meddat' data from dataset 
remove(meddat)
# Explanation:When the variable was written as its own (Group and Pop2003), the program failed recognize the given command. In order for the program to recognize what data I am referring to, I changed the variable name to "dataset$variable," giving a specific command on what dataset and variable the information should be retrieved from.

# Load data
load(url("http://www.jakebowers.org/Data/countries.rda"))
# Use 'dplyr' package
library(dplyr)
# Assign non-democratic value to "nonDemocraticCountries" variable
nonDemocraticCountries <- filter(smalldat, Cheibub4Type ??  "Democracy")
nonDemocraticCountries <- smalldat %>% filter(Cheibub4Type=="Dictatorship")
# Explanation: The coder wanted to select the nonDemocraticCountries from the data set. In order to do this, the %>% operator was used to pass data from one function to the next. Here, the smalldat data frame is filtered to pick out the "dictatorship" countries.  
# Assign democratic value to "democraticCountries" variable
democraticCountries <- filter(smalldat, Cheibub4Type %in% c("Democracy"))
democraticCountries <- filter(smalldat, Cheibub4Type %in% c("Parliamentary democracy", "Presidential democracy", "Mixed executive"))
# Explanation: In order to assign non-dictatorship countries to the variable democraticCountries, I changed the value "Democracy" to "Parliamentary democracy", "Presidential democracy" and "Mixed executive" as "democracy" alone does not exist in the data set.  
# Print the data assinged to these values 
print(nonDemocraticCountries)
print(democraticCountries)
# Remove datas from environment
remove(nonDemocraticCountries)
remove(democraticCountries)
remove(smalldat)

File: Buggy Code fixed [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# Out of these four possibilities, "Dictatrship" is the only to describe a non-democratic form of government. Thus, in order to create the non-democratic countries data frame, the appropriate code would be:
+nonDemocraticCountries <- smalldat %>% filter(Cheibub4Type=="Dictatorship")
+
+democraticCountries <- filter(smalldat, Cheibub4Type %in% c("Democracy"))
+# This code aims to create a data frame with democratic countries, but when the spreadsheet is created, the rows are empty. The 19 variables are there, but the observations are not. How do we fix this? As we learned with the previous code, "Democracy" is not a possibility, "Parliamentary democracy", "Mixed executive" and "Presidential democracy" are. Since we now have three possibilities to include in our data frame, we can't use the same command as before (Cheibub4Type==""). We need to use the combine values into a vector list function: c(). 
+help("c")
+# Then we also need to fix a bit the order and symbols (as in the previous case) for the filter function to work. The right code is:
+democraticCountries <- smalldat %>% filter(Cheibub4Type %in% c("Parliamentary democracy", "Mixed executive", "Presidential democracy"))
```
Comment (drenn2) [inline Buggy Code fixed]: There is an alternative way to do this using "!=" like this:

filter(Cheibub4Type !%in% c("Dictatorship") 

or simply

filter(Cheibub4Type != "Dictatorship")

File: Buggy Code fixed [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Buggy Code 1
+load(url("http://www.jakebowers.org/Data/meddat.rda"))
+colnames(meddat)
+# This function works and it shows the names of the variables (columns) in the meddat data frame
+
+table(Group, useNA = "ifany")
+# This function is suppossed to show the values for the "Group" variable, which has two possible outcomes (C-control group and T-treatment group). There are two errors with the code: (1) the name of the data frame was not included (meddat), and (2) the variable is named "nhGroup", not "Group". Here's the fixed code:
+table(meddat$nhGroup, useNA = "ifany")
```
Comment (drenn2) [inline Buggy Code fixed]: To clarify the function of git, if you just change the original line and save over the previous file I can see both the old line and your change. I suggest trying this next time (maybe even for the exploration)

**Comment by drenn2** [comment]
Thanks. Your fixes work. I'll make the next set more difficult

**Pull request closed without merging**